### DISCUSSION TITLE
How to control TextField only input the number?

### DISCUSSION THREAD

**Discussion by Ben2022-cell** [flet-dev/flet#951] [Q&A]
Hi，
    I require control the input text only is number, the code i setting the keyboard_type is number, but there can be input the number and other text, if has others solution or my code has some error?
`import flet as ft

def main(page: ft.Page):
    text = ft.Text()

    def change_text(e):
        text.value = e.control.value
        page.update()

    page.add(
        ft.Column(
            [
                ft.TextField(
                    label="额定电流",
                    keyboard_type=ft.KeyboardType.NUMBER,
                    on_change=change_text,
                ),
                text,
            ]
        )
    )

ft.app(target=main)`

**Comment by ndonkoHenri** [comment]
The Keyboard type is not meant for that. And also, your code has no error.
What you are asking isn't yet available. Many have tried to solve this issue by filtering the content of the TextField in the `on_change` callback.
If I find those discussions, I will share them here.

**Reply by Ben2022-cell** [reply]
thanks!

**Reply by ndonkoHenri** [reply]
Here is an attempt, on flet discord: https://discord.com/channels/981374556059086931/1055393769966538782
